server/planning/assignments/assignments_content.py

```diff
@@ -321,3 +323,26 @@ class AssignmentsContentResource(Resource):
     item_methods = []
 
     privileges = {"POST": "archive"}
+
+
+def merge_subject(item, planning):
+    vocabularies = get_resource_service("vocabularies").get_from_mongo(
+        req=None, lookup={"selection_type": "single selection"}, projection={"_id": 1}
+    )
+    single_value_vocabularies = set([v["_id"] for v in vocabularies])
+    subject = item.setdefault("subject", [])
+    if planning.get("subject"):
```

Comment:
Might be good to place this at the top of this function, so it doesn't needlessly lookup vocabs, and return if the Planning item doesn't have a subject